### PR TITLE
Add setfacl option --no-mask (#26796)

### PR DESCRIPTION
**Pull request by silvinux**
##### SUMMARY

- Added feature to not recalculate the effective rights mask by default (setfacl -n, --no-mask).

Issue #26796. 
module always recalculated the effective rights mask, which force mask permission over all files when it's recursive, and it's reflected on default permissions.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

acl.py

##### ANSIBLE VERSION

```
[root@silvinux ansible]# ansible --version
ansible 2.3.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jan 12 2017, 17:58:54) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
[root@silvinux15 ansible]# 


```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by silvinux** [comment]
needs_contributor

**Comment by silvinux** [comment]
@ansibot  needs_contributor

**Comment by silvinux** [comment]
needs_contributor

**Comment by silvinux** [comment]
ready_for_review

**Comment by silvinux** [comment]
@ansibot  ready_for_review

File: lib/ansible/modules/files/acl.py [hunk 1]
```diff
@@ -94,50 +94,44 @@
...
 
 EXAMPLES = '''
 # Grant user Joe read access to a file
-- acl:
-    path: /etc/foo.conf
-    entity: joe
-    etype: user
-    permissions: r
-    state: present
+- acl: name=/etc/foo.conf entity=joe etype=user permissions="r" state=present
```
Comment (alikins) [inline lib/ansible/modules/files/acl.py:115]: Why change the example docs? The yaml dict style is the preferred format for EXAMPLES

File: lib/ansible/modules/files/acl.py [hunk 2]
```diff
@@ -373,10 +368,12 @@ def main():
...
-        build_command(module, 'get', path, follow, default, recursive)
+        build_command(module, 'get', path, follow, default, recursive, no_mask)
     )
 
     module.exit_json(changed=changed, msg=msg, acl=acl)
 
-if __name__ == '__main__':
-    main()
+# import module snippets
+from ansible.module_utils.basic import *
```
Comment (alikins) [inline lib/ansible/modules/files/acl.py:377]: `from ansible.module_utils.basic import *` is deprecated now.

See http://docs.ansible.com/ansible/dev_guide/developing_modules_checklist.html

> Import ansible.module_utils code in the same place as you import other libraries. In older code, this was done at the bottom of the file but that’s no longer needed.

**Review by alikins — CHANGES_REQUESTED** [review]

**Comment by alikins** [comment]
ping @bcoca @astorije

**Comment by mattclay** [comment]
The test `ansible-test sanity --test pep8` failed with the following error:

```
test/sanity/pep8/legacy-files.txt:230:1: A201 Remove "lib/ansible/modules/files/acl.py" since it passes the current rule set
```

**Comment by VojtechMyslivec** [comment]
I have rebase it locally on v2.4.1.0-0.4.rc2 tag and it seems to work well.

@silvinux could you prepare the branch for merge please? I really need this feature.

Thanks!

**Comment by silvinux** [comment]
@VojtechMyslivec I'll take a look this week. I've forgot it.

**Comment by maxamillion** [comment]
Closing because changes were requested over a year ago and the PR seems to be abandoned. Please feel free to reopen if you desire to continue this work.

**Pull request closed without merging**